### PR TITLE
fix: System user locked.

### PR DESCRIPTION
**Pull request by Ash-exp**
# Description

After installing Devtron, initially 2 users are there. We have marked 'admin' user as locked.
Since, we cannot update permissions for 'system' users so it would be better to mark system user also as locked.

Ticket [AB1664](https://dev.azure.com/DevtronLabs/Devtron/_boards/board/t/OSS%20adoption/Stories/?workitem=1664)
Fixes https://github.com/devtron-labs/devtron/issues/2832

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- Navigate to global configurations
- Go to Authorization -> User Permission
- `system` user should be locked just as `admin`.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

Screenshot 2023-02-06 at 6 57 28 PM

### CODE REVIEW

File: src/components/userGroups/UserGroup.tsx [hunk 1]
```diff
@@ -627,6 +627,14 @@ const CollapsedUserOrGroup: React.FC<CollapsedUserOrGroupProps> = ({
         setData((state) => ({ ...state, result: data }))
         updateCallback(index, data)
     }
+
+    function getToolTipContent(user) {
+        switch (user) {
+            case 'admin': return "Admin user cannot be edited"
```
Comment (shivani170) [inline src/components/userGroups/UserGroup.tsx:633]: Can we keep this admin and system in the enum??

File: src/components/userGroups/UserGroup.tsx [hunk 2]
```diff
@@ -639,23 +647,23 @@ const CollapsedUserOrGroup: React.FC<CollapsedUserOrGroupProps> = ({
                 </span>
                 <span
                     className="user-list__direction-container flex rotate pointer"
-                    onClick={email_id === 'admin' ? noop : (e) => setCollapsed(not)}
+                    onClick={email_id === 'admin' || email_id === 'system' ? noop : (e) => setCollapsed(not)}
```
Comment (shivani170) [inline src/components/userGroups/UserGroup.tsx:650]: Extract out this logic in constant

File: src/components/userGroups/UserGroup.tsx (hunk 1 above)
Comment (Ash-exp) [reply src/components/userGroups/UserGroup.tsx]: updated 👍

File: src/components/userGroups/UserGroup.tsx (hunk 2 above)
Comment (Ash-exp) [reply src/components/userGroups/UserGroup.tsx]: updated 👍

**Review by shivani170 — APPROVED** [review]

File: src/components/userGroups/UserGroup.tsx [hunk 3]
```diff
@@ -636,6 +637,16 @@ const CollapsedUserOrGroup: React.FC<CollapsedUserOrGroupProps> = ({
         updateCallback(index, data)
     }
 
+    function getToolTipContent(user) {
+        switch (user) {
+            case UserLocked.ADMIN :
+                return 'Admin user cannot be edited'
```
Comment (shivani170) [inline src/components/userGroups/UserGroup.tsx:643]: Can we extract these messaging from constantMessaging file

Comment (Ash-exp) [reply src/components/userGroups/UserGroup.tsx]: implemented 👍

File: src/components/userGroups/UserGroup.tsx [hunk 4]
```diff
@@ -636,6 +637,16 @@ const CollapsedUserOrGroup: React.FC<CollapsedUserOrGroupProps> = ({
...
+            case UserLocked.SYSTEM :
+                return 'System user cannot be edited'
```
[the remark this answers is not in the document]
Comment (Ash-exp) [reply src/components/userGroups/UserGroup.tsx]: implemented 👍

File: src/components/userGroups/UserGroup.tsx [hunk 5]
```diff
@@ -636,6 +637,16 @@ const CollapsedUserOrGroup: React.FC<CollapsedUserOrGroupProps> = ({
         updateCallback(index, data)
     }
 
+    function getToolTipContent(user) {
+        switch (user) {
```
Comment (shivani170) [inline src/components/userGroups/UserGroup.tsx:641]: Can try with if else instead switch

File: src/config/constantMessaging.ts [hunk 6]
```diff
@@ -140,3 +140,9 @@ export const APP_GROUP_CI_DETAILS = {
         subTitle: 'Pipeline trigger history, details and logs will be available here.',
     },
 }
+
+// user locked message
+export const USER_LOCK_MESSAGE = {
```
Comment (shivani170) [inline src/config/constantMessaging.ts:145]: Please remove this const and utilise existing one for admin/system

File: src/components/userGroups/UserGroup.tsx (hunk 5 above)
Comment (Ash-exp) [reply src/components/userGroups/UserGroup.tsx]: Implemented 👍

**Review by shivani170 — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**